### PR TITLE
Feature/#607 - Add `validity_period` attribute to DataNodeConfig

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy-core/issues/607

### CODE REVIEW

File: src/taipy/core/config/checkers/_data_node_config_checker.py [hunk 1]
```diff
@@ -60,6 +62,15 @@ def _check_scope(self, data_node_config_id: str, data_node_config: DataNodeConfi
                 f"`{Scope.PIPELINE}` is deprecated. Please use other `Scope` value instead.",
             )
 
+    def _check_validity_period(self, data_node_config_id: str, data_node_config: DataNodeConfig):
+        if data_node_config.validity_period and not isinstance(data_node_config.validity_period, timedelta):
+            self._error(
+                data_node_config._VALIDITY_PERIOD_KEY,
+                data_node_config.validity_period,
+                f"`{data_node_config._VALIDITY_PERIOD_KEY}` field of DataNodeConfig `{data_node_config_id}` must be"
+                f" populated with a timedelta value.",
```
Comment (jrobinAV) [inline src/taipy/core/config/checkers/_data_node_config_checker.py]:
```suggestion
                f" None or populated with a timedelta value.",
```

File: tests/core/config/test_config.py [hunk 2]
```diff
@@ -80,7 +81,26 @@ def test_configure_sql_data_node(self):
...
+            "db_name",
+            "collection_name",
+            None,
+            "user",
+            "pwd",
+            "host",
+            "port",
+            {"foo": "bar"},
+            Scope.PIPELINE,
+            timedelta(1),
```
Comment (jrobinAV) [inline tests/core/config/test_config.py]: Good catch. It was missing.

File: tests/core/config/test_config.py [hunk 3]
```diff
@@ -80,7 +81,26 @@ def test_configure_sql_data_node(self):
...
+            "foo",
+            "db_name",
+            "collection_name",
+            None,
+            "user",
+            "pwd",
+            "host",
+            "port",
+            {"foo": "bar"},
+            Scope.PIPELINE,
```
Comment (jrobinAV) [inline tests/core/config/test_config.py]:
```suggestion
            Scope.SCENARIO,
```

To avoid future merge issues

**Review by jrobinAV — DISMISSED** [review]
minor remarks

**Review by toan-quach — DISMISSED** [review]
lgtm

File: tests/core/config/checkers/test_data_node_config_checker.py [hunk 4]
```diff
@@ -268,6 +269,43 @@ def test_check_scope(self, caplog):
...
+        assert expected_error_message in caplog.text
+
+        config._sections[DataNodeConfig.name]["default"].validity_period = 1
+        with pytest.raises(SystemExit):
+            Config._collector = IssueCollector()
+            Config.check()
+        assert len(Config._collector.errors) == 1
+        expected_error_message = (
+            "`validity_period` field of DataNodeConfig `default` must be populated with a timedelta"
+            " value. Current value of property `validity_period` is 1."
```
Comment (trgiangdo) [inline tests/core/config/checkers/test_data_node_config_checker.py]:
```suggestion
            "`validity_period` field of DataNodeConfig `default` must be populated with None or"
            " a timedelta value. Current value of property `validity_period` is 1."
```

File: tests/core/config/checkers/test_data_node_config_checker.py [hunk 5]
```diff
@@ -268,6 +269,43 @@ def test_check_scope(self, caplog):
...
+        config = Config._default_config
+
+        config._sections[DataNodeConfig.name]["default"].validity_period = "bar"
+        with pytest.raises(SystemExit):
+            Config._collector = IssueCollector()
+            Config.check()
+        assert len(Config._collector.errors) == 1
+        expected_error_message = (
+            "`validity_period` field of DataNodeConfig `default` must be populated with a timedelta"
+            ' value. Current value of property `validity_period` is "bar".'
```
Comment (trgiangdo) [inline tests/core/config/checkers/test_data_node_config_checker.py]:
```suggestion
            "`validity_period` field of DataNodeConfig `default` must be populated with None or"
            ' a timedelta value. Current value of property `validity_period` is "bar".'
```

File: tests/core/config/checkers/test_data_node_config_checker.py [hunk 6]
```diff
@@ -268,6 +269,43 @@ def test_check_scope(self, caplog):
...
+    def test_check_validity_period(self, caplog):
+        config = Config._default_config
+
+        config._sections[DataNodeConfig.name]["default"].validity_period = "bar"
+        with pytest.raises(SystemExit):
+            Config._collector = IssueCollector()
+            Config.check()
+        assert len(Config._collector.errors) == 1
+        expected_error_message = (
+            "`validity_period` field of DataNodeConfig `default` must be populated with None or"
```
Comment (trgiangdo) [inline tests/core/config/checkers/test_data_node_config_checker.py]:
```suggestion
            "`validity_period` field of DataNodeConfig `default` must be None or populated with"
```

File: tests/core/config/checkers/test_data_node_config_checker.py [hunk 7]
```diff
@@ -268,6 +269,43 @@ def test_check_scope(self, caplog):
...
+        )
+        assert expected_error_message in caplog.text
+
+        config._sections[DataNodeConfig.name]["default"].validity_period = 1
+        with pytest.raises(SystemExit):
+            Config._collector = IssueCollector()
+            Config.check()
+        assert len(Config._collector.errors) == 1
+        expected_error_message = (
+            "`validity_period` field of DataNodeConfig `default` must be populated with None or"
```
Comment (trgiangdo) [inline tests/core/config/checkers/test_data_node_config_checker.py]:
```suggestion
            "`validity_period` field of DataNodeConfig `default` must be None or populated with"
```

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**